src/Teams/Models/IRepository.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Teams.Models
+{
+    interface IRepository<T> where T : class
+    {
+        IEnumerable<T> GetAll();
+        T GetById(object id);
```

Comment:
why object? If the id will be of value type then you will trigger boxing/unboxing mechanism. You can return IQueryable from the interface or add an additional generic parameter as a key